base/sparse/sparsematrix.jl

```diff
@@ -2265,7 +2265,15 @@ function spset!{Tv,Ti<:Integer}(A::SparseMatrixCSC{Tv}, x::Tv, I::AbstractVector
 
     m, n = size(A)
     lenI = length(I)
-    ((I[end] > m) || (J[end] > n)) && throw(DimensionMismatch(""))
+
+    if (!isempty(I) && (I[1] < 1 || I[end] > m)) || (!isempty(J) && (J[1] < 1 || J[end] > n))
+        throw(BoundsError(A, (I, J)))
+    end
+
+    if isempty(I) || isempty(J)
+        return A
```

Comment:
And the functions in question here return the LHS at the end of their normal code path, so I thought I better stick to that.